src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -480,88 +481,154 @@ void mongoSubCacheRefresh(const std::string& database)
+*
+* mongoSubCountersUpdateLastFailure - 
+*/
+static void mongoSubCountersUpdateLastFailure
+(
+  const std::string&  collection,
+  const std::string&  subId,
+  long long           lastFailure
+)
+{
+  BSONObj      condition;
+  BSONObj      update;
+  std::string  err;
+
+  condition = BSON("_id" << OID(subId) << "$or" << BSON_ARRAY(
+                     BSON(CSUB_LASTFAILURE << BSON("$lt" << lastFailure)) <<
+                     BSON(CSUB_LASTFAILURE << BSON("$exists" << false))));
+  update    = BSON("$set" << BSON(CSUB_LASTFAILURE << lastFailure));
+
+  if (collectionUpdate(collection, condition, update, false, &err) != true)
   {
-    // Update lastFailure    
-    condition = BSON("_id" << OID(subId) << "$or" << BSON_ARRAY(
-                       BSON(CSUB_LASTFAILURE << BSON("$lt" << lastFailure)) <<
-                       BSON(CSUB_LASTFAILURE << BSON("$exists" << false)))
-                    );
-    update    = BSON("$set" << BSON(CSUB_LASTFAILURE << lastFailure));
-
-    if (collectionUpdate(collection, condition, update, false, &err) != true)
-    {
-      LM_E(("Internal Error (error updating 'lastFailure' for a subscription)"));
-    }
+    LM_E(("Internal Error (error updating 'lastFailure' for a subscription)"));
   }
-  else
+}
+
+
+
+/* ****************************************************************************
+*
+* mongoSubCountersUpdateLastSuccess - 
+*/
+static void mongoSubCountersUpdateLastSuccess
+(
+  const std::string&  collection,
+  const std::string&  subId,
+  long long           lastSuccess
+)
+{
+  BSONObj      condition;
+  BSONObj      update;
+  std::string  err;
+
+  condition = BSON("_id" << OID(subId) << "$or" << BSON_ARRAY(
+                     BSON(CSUB_LASTSUCCESS << BSON("$lt" << lastSuccess)) <<
+                     BSON(CSUB_LASTSUCCESS << BSON("$exists" << false))));
+  update    = BSON("$set" << BSON(CSUB_LASTSUCCESS << lastSuccess));
+
+  if (collectionUpdate(collection, condition, update, false, &err) != true)
   {
-    // Set lastFailure to -1
-    condition = BSON("_id" << OID(subId));
-    update    = BSON("$set" << BSON(CSUB_LASTFAILURE << (long long) -1));
+    LM_E(("Internal Error (error updating 'lastSuccess' for a subscription)"));
+  }
+}
 
-    if (collectionUpdate(collection, condition, update, false, &err) != true)
-    {
-      LM_E(("Internal Error (error updating 'lastFailure' for a subscription)"));
-    }
+
+
+/* ****************************************************************************
+*
+* mongoSubCountersUpdate - update subscription counters and timestamps in mongo
+*
+* FIXME P4:
+*   This function the way it is implemented right now might perform FIVE consecutive
+*   updates in mongo.
+*   There must be some faster way to do this:
+*     1. Read first, update only what's necessary OR
+*     2. Do a 'mega update' with all five fields (have the functions concatenate to final mongo 'query')
```

Comment:
I was thinking on this at the very begining, to "package" count and lastNofication updates into just one operation instead 2...

The fact is that given the filtering condition for each update needs to be evaluated "locally" for each lastX field, you need to split in several places. 

Thus, probably this FIXME should be removed.